common/crypto/platform/openssl/crypto.cpp

```diff
@@ -472,11 +571,17 @@ expected::ExpectedBool VerifySignData(
 	ret = EVP_PKEY_verify(
 		pkey_signer_ctx.get(), signature.data(), signature.size(), shasum.data(), shasum.size());
 	if (ret < 0) {
+		const string openssl_error_msg {GetOpenSSLErrorMessage()};
+		if (openssl_error_msg.find("asn1 encoding") != string::npos) {
+			log::Debug(
+				"Failed to verify the signature with the supported OpenSSL binary formats. Falling back to the custom Mender encoded binary format for ECDSA signatures");
+			return VerifyECDSASignData(public_key_path, shasum, signature);
+		}
 		return expected::unexpected(MakeError(
 			VerificationError,
-			"Failed to verify signature. OpenSSL PKEY verify failed: " + GetOpenSSLErrorMessage()));
+			"Failed to verify the new der encoded signature. OpenSSL PKEY verify failed: "
```

Comment:
A bit difficult to read, what does "new" mean in this context? The signature did not validate and we didn't try the fallback method neither. At least use `DER` instead:
```suggestion
			"Failed to verify the DER encoded signature. OpenSSL PKEY verify failed: "
```

But also, it should be `DER/PEM` ?